### DISCUSSION TITLE
Use Open WebUI at a large-scale

### DISCUSSION THREAD

**Discussion by OM-EL** [open-webui/open-webui#7771] [General]
A lot of people use OWUI for personal use. However, different types of limitations/problems appear when deploying OWUI on a large scale (1000 users, 200 requests/second and more for example ). In this discussion, we will address problems related to scaling up the usage of OWUI.

**Comment by nboehlke** [comment]
Imagine having the same question, searching for it, and then finding someone else who asked the exact same thing just a couple of hours ago. I love it.

I’m using OWUI for personal purposes as well, and I’m trying to get this system up and running for a limited number of users in my company - just to prove that there are alternatives to BigAi.

I’d like to know if anyone else out there has had the same idea and encountered any limitations. Also, which hardware are you using, and on what system are you running it?

For a proof of concept, I’d start with high-end consumer hardware before scaling up to professional equipment like the H200, and so on.

**Reply by OM-EL** [reply]
I will be happy to share with you the structure i used to build the system.
For now i am making use of Bedrock , the goal is to keep the possibility to move back full on premise whenever they have the hardware.

![Sans-titre-2024-12-11-1328](https://github.com/user-attachments/assets/27c28eb9-1da5-4a27-b3c2-6ec973a64f9d)

**Reply by nboehlke** [reply]
Didn’t expect an architectural ramp-up—wow! Thank you so much for sharing!
It’s more than I’d like to implement, but it’s always good to gain new perspectives.

Regarding hardware, I’m considering setting up a Red Hat OpenShift environment by the time NVIDIA ships the 5090. As I mentioned, it should be sufficient for a proof of concept (especially when looking at the cost comparison between consumer and enterprise hardware).

Since this will be a PoC, I’d like to showcase as much as possible on that hardware. This Git repository has a satisfactory set of examples I’d like to run: https://github.com/CastawayEGR/openshift-ai-examples

My anticipated hardware setup is:

CPU:
AMD Ryzen 9 9900X, 12 cores, 5.6 GHz, AM5 (Zen 5)

GPU:
2 x Nvidia RTX 5090 32GB (no NVLink—no unified VRAM!)

Mainboard:
ProArt X670E-CREATOR WIFI

RAM:
2 x 32GB Kingston Server Premier DDR5-4800 DIMM CL40 (ECC RAM)
KSM48E40BD8KM-32HM

Storage:
Crucial T700 NVMe SSD, PCIe 5.0 M.2 2280 - 2 TB

Power Supply:
be quiet! Straight Power 12, 80 PLUS Platinum, ATX 3.0, PCIe 5.0 - 1500 Watt
(Upgrade to 1500+ watts if GPU or CPU are overclocked)

Case:
PHANTEKS Enthoo Pro 2 Server Big-Tower, XL-EEB - black

**Reply by dangal95** [reply]
How does this perform? I have something similar running but I am experiencing sporaddic slow chat response times at Scale. I run Open WebUI (front-end and back-end in one container, is it possible to separate them?), Open WebUI Pipelines, and LiteLLMProxy as separate ECS Services. Open WebUI is exposed via an Application Load Balancer (ALB), and it autoscales from 1 Task to 5 Tasks depending on the request per target rate that the ALB registers. Each instance has 4 uvicorn workers, 8 vCPU, and 16GB ram. Open WebUI Pipelines and LiteLLMProxy do not autoscale but they are also not the problem. I've tested this by connecting from Open WebUI directly to Azure OpenAI and I still had the same problem which I will discuss below.

Open WebUI uses an Aurora PostgreSQL Cluster (one writer and one reader) as its DB backend, and Redis (not in Cluster Mode but has one primary and one replica) to both store information and handle websockets. I use Azure OpenAI (via LiteLLM), so I am not running any local models.

All metrics for all Services and Databases on AWS look good. I've even connected to Redis and ran a few commands to make sure that it is being used for websockets, the number of connections, blocked connections etc and everything looks good. The response time from Azure OpenAI is pretty fast too, and LiteLLM relays back almost immediately to Open WebUI.


Yet, at scale (let's say around 1000 users too) the UI sporadically slows down. By this I mean that the UI is accessible and you can click around, but when you send a message to an LLM via Chat it first takes a long time (say 10-20 seconds) for the "sending" action to occur, then once the response is received (I am monitoring the Network tab in chrome when opening Developer mode) it is printed at a really slow rate. The rate is around 1 token per 2 seconds, so it seems to be lagging. I really suspected it was a websocket issue but after my analysis I think I can rule that out. 

Has anyone had this problem? If so how did you solve it? If you did not run into it, do you perhaps have a hunch on what it could be, given what I just explained? Any help is appreciated 🙏

**Comment by mahenning** [comment]
Currently I'm in the process of preparing Open-WebUI for my organisation (~1000 employees). I can't yet speak about performance (problems), but I chose to go with Open-WebUI + vLLM (on an A100), as vLLM is more powerful than ollama when it comes to parallel requests from multiple users due to batched inference. 200 requests/second will be a bit much even for vLLM, but it will be much faster than ollama on that. Although with 1000 users, 200 req/sec is a bit much.

Potential problems/work for me:
* No "I forgot my password" functionality. Right now, if someone forgets their password, they have to write me (or another admin) a mail and we have to reset the password manually + send out the new password.
* No mail address check. Users can register with whatever name/e-mail they want, and it is up to the admins to enforce proper account naming to identify users. In the worst case, a user could potentially use the name and e-mail address of another person in the company to register and fake being the person in the WebUI for the admins.

I would be interested what kind of limitations/problems you encountered, if any yet.

**Reply by smonux** [reply]
I haven't used it, but OpenWebUI supports LDAP. I think that would be a better solution than letting every user input their corporate email.

**Reply by mahenning** [reply]
Yes thanks for the tip! I saw the option just yesterday. I have to talk to my org but this would be a great way to make it easier for both sides.

**Reply by derek-assurity** [reply]
Look at putting a Auth Proxy in front of Open WebUI and have it manage access via SSO using Trusted Headers. I don't recommend OWUI integration with SSO, or using OAuthProxy. Envoy Proxy is solid and handles this pretty well, just need to be careful as there are some gotchas with expected token availability. The section on [Trusted Headers](https://docs.openwebui.com/features/sso#trusted-header) will point you in the right direction.

**Comment by jk-f5** [comment]
I'm currently working on exactly this. My issue with scaling multiple replicas of the openwebui container alongside redis is addressed here: #7887
